geomesa-kafka/geomesa-kafka-datastore/src/test/scala/org/locationtech/geomesa/kafka/utils/GraphVizFilterVisitor.scala

```diff
@@ -0,0 +1,144 @@
+/***********************************************************************
+* Copyright (c) 2013-2016 Commonwealth Computer Research, Inc.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Apache License, Version 2.0
+* which accompanies this distribution and is available at
+* http://www.opensource.org/licenses/apache2.0.php.
+*************************************************************************/
+
+package org.locationtech.geomesa.kafka.utils
+
+import org.opengis.filter._
+import org.opengis.filter.spatial._
+import org.opengis.filter.temporal._
+
+import scala.collection.JavaConversions._
+
+class GraphVizFilterVisitor extends FilterVisitor {
+  override def visit(filter: And, extraData: scala.Any): AnyRef = {
+    val count = extraDataToCount(extraData)
+
+    printOperand(filter, "AND", count)
+    //filter.getChildren.foreach(_.accept(this, count + 1))
+    linkChildren(filter, count)
+  }
+
+  override def visit(filter: Or, extraData: scala.Any): AnyRef = {
+    val count = extraDataToCount(extraData)
+
+    printOperand(filter, "OR", count)
+    //filter.getChildren.foreach(_.accept(this, count + 1))
+    linkChildren(filter, count)
+  }
+
+  def linkChildren(binary: BinaryLogicOperator, count: Int) = {
+    binary.getChildren.foreach { c =>
+      val childRand = math.abs(scala.util.Random.nextInt())
+      println(s"node_${nodeName(binary, count)} -> node_${nodeName(c, childRand)}")
```

Comment:
Printing out Graphviz code is arguably the point of the whole class. But as @jnh5y says above, it probably doesn't belong in this PR.